src/main/java/build/buildfarm/common/services/WriteStreamObserver.java

```diff
@@ -291,6 +291,12 @@ private boolean errorResponse(Throwable t) {
                 requestMetadata.getToolInvocationId(),
                 requestMetadata.getActionId(),
                 name));
+      } else if ("write cancelled".equals(t.getMessage())) {
```

Comment:
Prefer that we get a WriteException (`extends IOException`) that specializes to a WriteCancelledException

Reply:
We were also getting the same error. This pr has fixed the issue for us https://github.com/bazelbuild/bazel-buildfarm/pull/1526.